### ISSUE TITLE
Work around librdkafka DNS limitation

### ISSUE THREAD

**Issue by mcandre** [confluentinc/confluent-kafka-go#49]
Could confluent-kafka-go guard against mDNS failures in the librdkafka code path, which result in `.local`-style broker addresses failing to connect in some environments?

One way to do this is to resolve the configured bootstrap server addresses, using the pure Go resolver, before handing this configuration to librdkafka.

**Comment by edenhill** [comment]
Do note though that librdkafka will resolve the broker names as they are reported from the brokers themselves (advertised.host.name).

**Comment by edenhill** [comment]
Are your brokers return .local hostnames as their advertised hostnames, or are you supplying bootstrap.servers with .local hostnames?

**Comment by mcandre** [comment]
After more tinkering, I learned that my build command was wrong--I was trying to specify the compiler and linker flags directly from go build, but as the README indicates, the correct call is `go build -tags static`. Once I did it that way, DNS started working, including mDNS .local addresses.

For what it's worth, I'm using musl-dev in alpine, rather than glibc. Don't know / don't care whether that matters.
